doc/man7/provider-cipher.pod

```diff
@@ -148,7 +148,9 @@ It is the responsibility of the cipher implementation to handle input lengths
 that are not multiples of the block length.
 In such cases a cipher implementation will typically cache partial blocks of
 input data until a complete block is obtained.
-I<out> may be the same location as I<in> but it should not partially overlap.
+The pointers I<out> and I<in> may point to the same location, in which
+case the encryption must be done in-place. If they point to different locations,
+the operation must fail if the buffers overlap partially.
```

Comment:
I think `is expected to fail` is better. But I am not even sure we currently check this for all ciphers? And I do not think it is reasonable to ask for. There can be third party providers that don't and imposing further requirements is breaking backwards compatibility with them.

Reply:
So no, if I did not overlook something, we do not currently check this in provider code - only in legacy codepaths.